base/entries.go

```diff
@@ -0,0 +1,63 @@
+package base
+
+import (
+	"fmt"
+
+	"github.com/qri-io/dataset/dsio"
+)
+
+// ReadEntries reads entries and returns them as a native go array or map
+func ReadEntries(reader dsio.EntryReader, all bool, limit int, offset int) (interface{}, error) {
+	obj := make(map[string]interface{})
+	array := make([]interface{}, 0)
+	numRead := 0
+
+	tlt, err := dsio.GetTopLevelType(reader.Structure())
+	if err != nil {
+		return nil, err
+	}
+
+	for i := 0;; i++ {
+		val, err := reader.ReadEntry()
+		if err != nil {
+			if err.Error() == "EOF" {
+				break
+			}
+			return nil, err
+		}
+		if !all && i < offset {
+			continue
+		}
+
+		if tlt == "object" {
+			obj[val.Key] = val.Value
+		} else {
+			array = append(array, val.Value)
+		}
+
+		numRead++
+		if !all && numRead == limit {
+			break
+		}
+	}
+
+	if tlt == "object" {
+		return obj, nil
+	}
+	return array, nil
+}
+
+// ReadEntriesToArray reads entries and returns them as a native go array
+func ReadEntriesToArray(reader dsio.EntryReader, all bool, limit int, offset int) ([]interface{}, error) {
```

Comment:
On the other hand I'm not a fan of this. I think this is prone to misuse, or leads the reader to falsely assume we don't/rarely return objects. 

Since this is only being used in testing, I think we should change this to a `Must` function, because it seems the main purpose for using this is within the context of tests:
```go
// MustSliceEntries asserts the returned entries are a slice of interface{}
// it panics if any other value or a non-nil error is returned
// this is mainly intended for use in tests
func MustSliceEntries(ent interface{}, err error) []interface{} {
  if err != nil {
    panic(err)
  }
 entSlice, ok := ent.([]interface{})
  if !ok {
    panic("entries is not of type []interface{}")
  } 
  return entSlice
}
```
use it like this: 
```go
  ent := MustSliceEntries(ReadEntries(rdr, all, limit, offset))
```

Reply:
Good call. Moved the function to lib/datasets_test (where it is needed) and called it `mustByArray`.